### PR TITLE
[flutter_markdown] Add TableCellVerticalAlignment property in markdown stylesheet

### PR DESCRIPTION
**Pull request by joshpetit**
Add property  `tableVerticalAlignment` to markdown style sheet that sets the vertical alignment of table properties.

flutter/flutter#125872

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Comment by joshpetit** [comment]
I see this was approved but still needs to be merged in. Is there any ETA on that? I'm planning on using this fix in a release and am wondering if I can wait till this is merged or if it'll take some more time.

**Comment by stuartmorgan** [comment]
> I see this was approved but still needs to be merged in.

It needs a second review, per https://github.com/flutter/flutter/wiki/Tree-hygiene#getting-a-code-review. It's in my queue.

File: packages/flutter_markdown/CHANGELOG.md
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:3]: Adds

See the changelog style guide linked from the PR checklist.

Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:3]: Please add backticks around the code reference.

File: packages/flutter_markdown/lib/src/style_sheet.dart
Comment (stuartmorgan) [inline packages/flutter_markdown/lib/src/style_sheet.dart:704]: Why is this line duplicated?

File: packages/flutter_markdown/test/table_test.dart
Comment (stuartmorgan) [inline packages/flutter_markdown/test/table_test.dart:125]: Isn't this only testing one value? If so, then everything would pass even if someone changed it to just always use `top` the way it used to always use `middle`.

You also don't have any testing of the default handling, unless I missed it.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/flutter_markdown/test/table_test.dart
Comment (stuartmorgan) [inline packages/flutter_markdown/test/table_test.dart:164]: Two different tests should not have the same name; these need to distinguish what is different between the tests.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Looks good with one small change, sorry for the delay.

**Comment by stuartmorgan** [comment]
@joshpetit Are you still planning on updating this to resolve the last issue? Once that's done it should be good to land.

**Comment by joshpetit** [comment]
Oh thanks for the ping, I'll look at your comments tonight.

**Pull request merged**
